lib/common/JSONImporter/NodeChangeSet.ts

```diff
@@ -25,4 +27,17 @@ export class NodeChangeSet implements ChangeSet {
             diffObj.value
         );
     }
+
+    validated(fn: (ch: NodeChangeSet) => boolean, errMsg: string): Result<NodeChangeSet, Error> {
```

Comment:
If we are passing the validation function here (which is only called once per patch operation right before applying it), it probably doesn't really buy us much defining this here. I know I initially suggested something like this but it might make more sense to just define a validation function for patch operations. I made an example of what I mean in a commit and pushed it.